### PR TITLE
docs: Add git workflow to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by samwho**
#### Description

When I submitted my first PR to this project the other, I wasn't sure if the way I had used git would be acceptable. Some open source project are quite specific about the way they want you to use git. Fortunately it was fine, but I thought it would be a nice idea to be explicit about it in the contributing guidelines. 🙂 

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### How Has This Been Tested?
- [ ] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:
- [x] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -99,3 +99,13 @@ $ npm run dev
...
+
+### Git/GitHub workflow
+
+1. Fork this repository
+2. Create a branch off of `master` for your work: `git checkout -b my-feature-branch`
+3. Make some changes, committing them along the way
+4. When your changes are ready for review, push your branch: `git push origin my-feature-branch`
+5. Create a pull request from your branch to `starship/master`
+6. No need to assign the pull request to anyone, we'll review it when we can
+7. When the changes have been reviewed and approved, someone will merge it
```
Comment (chipbuster) [inline CONTRIBUTING.md]: Would be nice to add that the merge process will automatically squash and rebase, so there's no need to do that manually on the PR.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -99,3 +99,13 @@ $ npm run dev
 ```
 
 Once setup is complete, you can refer to VuePress documentation on the actual implementation here: https://vuepress.vuejs.org/guide/.
+
+### Git/GitHub workflow
+
```
Comment (chipbuster) [inline CONTRIBUTING.md]: Could we have a one-line intro to the section here? Just something to the effect of "this is our preferred process for how to make a PR on github"

File: CONTRIBUTING.md (hunk 1 above)
Comment (samwho) [reply CONTRIBUTING.md]: Good call! Made a small edit to make this more clear. Let me know if you think it should be even more explicit. :)

File: CONTRIBUTING.md (hunk 2 above)
Comment (samwho) [reply CONTRIBUTING.md]: Absolutely, done.

**Pull request merged**